storage/container.go

```diff
@@ -37,18 +41,60 @@ func (c *Container) GetURL() string {
 	return c.bsc.client.getEndpoint(blobServiceName, pathForResource(container, ""), nil)
 }
 
-func (c *Container) GetSASURI(expiry time.Time, permissions string) (string, error) {
-	return c.GetSASURIWithSignedIPAndProtocol(expiry, permissions, "", false)
+// ContainerSASOptions are options to construct a container SAS
+// URI.
+// See https://docs.microsoft.com/en-us/rest/api/storageservices/constructing-a-service-sas
+type ContainerSASOptions struct {
+	ContainerSASPermissions
+	OverrideHeaders
+	SASOptions
 }
 
-func (c *Container) GetSASURIWithSignedIPAndProtocol(expiry time.Time, permissions string, signedIPRange string, HTTPSOnly bool) (string, error) {
+// ContainerSASPermissions includes the available permissions for
+// a container SAS URI.
+type ContainerSASPermissions struct {
+	Read   bool
+	Add    bool
+	Create bool
+	Write  bool
+	Delete bool
+	List   bool
+}
+
+// GetSASURI creates an URL to the container which contains the Shared
+// Access Signature with the specified options.
+//
+// See https://docs.microsoft.com/en-us/rest/api/storageservices/constructing-a-service-sas
+func (c *Container) GetSASURI(options ContainerSASOptions) (string, error) {
 	uri := c.GetURL()
 	signedResource := "c"
 	canonicalizedResource, err := c.bsc.client.buildCanonicalizedResource(uri, c.bsc.auth)
 	if err != nil {
 		return "", err
 	}
-	return c.bsc.client.commonSASURI(expiry, uri, permissions, signedIPRange, canonicalizedResource, signedResource, HTTPSOnly)
+
+	// build permissions string
+	permissions := ""
```

Comment:
Move this logic into a base type and make it a method so you don't have to repeat it.